actors/evm/src/interpreter/memory.rs

```diff
@@ -16,12 +16,15 @@ impl Default for Memory {
 
 impl Memory {
     #[inline]
+    /// Resizes memory to `size` length, reserving extra WASM pages as-needed.
+    /// TODO this should be renamed resize since it can also shrink memory.
```

Comment:
Yeah, this is weird. Ideally:

1. This method would grow only.
2. We'd change the call sites to not check the size (and just blindly call this method to grow when necessary.

Reply:
We have a redundant `grow_memory` method that should probably be merged into this assoc which does this for us currently. 

We should still pass _some_ size because need to differentiate between a page growth and a word grow.

```rust 
fn reserve_pages(&mut self, pages: usize)

pub fn grow(&mut self, new_len: usize) {
     // <self.reserve_pages used as-needed>
}
```
where new_len is not expected to be aligned so we do the alignment inside `grow`.